collect_app/src/main/java/org/odk/collect/android/google/GoogleAccountsManager.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (C) 2017 Shobhit
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.odk.collect.android.google;
+
+
+import android.Manifest;
+import android.app.Activity;
+import android.app.Fragment;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentSender;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GooglePlayServicesUtil;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.api.client.extensions.android.http.AndroidHttp;
+import com.google.api.client.googleapis.extensions.android.gms.auth.GoogleAccountCredential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.ExponentialBackOff;
+import com.google.api.services.drive.DriveScopes;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+import org.odk.collect.android.preferences.PreferenceKeys;
+import org.odk.collect.android.utilities.ToastUtils;
+
+import java.util.Collections;
+import java.util.List;
+
+import pub.devrel.easypermissions.EasyPermissions;
+import timber.log.Timber;
+
+public class GoogleAccountsManager implements EasyPermissions.PermissionCallbacks,
+        GoogleApiClient.OnConnectionFailedListener {
+
+    public static final int REQUEST_ACCOUNT_PICKER = 1000;
+    private static final int REQUEST_PERMISSION_GET_ACCOUNTS = 1002;
+    private static final int RESOLVE_CONNECTION_REQUEST_CODE = 5555;
+
+    @Nullable
+    private Fragment fragment;
+    @Nullable
+    private Activity activity;
+    @Nullable
+    private GoogleAccountSelectionListener listener;
+
+    private Context context;
+    private GoogleAccountCredential credential;
+
+    private DriveHelper driveHelper;
+    private SheetsHelper sheetsHelper;
+
+    private HttpTransport transport;
+    private JsonFactory jsonFactory;
```

Comment:
There are technically code paths where these are not initialized (the testing-only constructor). Technically as long as that constructor is only used for testing you're fine, but I'd still probably like to see @Nullables on these.

Reply:
:+1: